compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -96,12 +96,12 @@ pub fn intern_const_alloc_recursive<
 ) -> Result<(), ErrorGuaranteed> {
     // We are interning recursively, and for mutability we are distinguishing the "root" allocation
     // that we are starting in, and all other allocations that we are encountering recursively.
-    let (base_mutability, inner_mutability) = match intern_kind {
+    let (base_mutability, inner_mutability, is_static) = match intern_kind {
```

Comment:
The doc comment for this function should explain that if `intern_kind` is a `Static`, then the root allocation will *not* be interned and it is up to the caller to deal with that.